lib/vdsm/virt/qemuguestagent.py

```diff
@@ -781,3 +795,24 @@ def _qga_call_get_devices(self, vm):
         else:
             self.set_failure(vm.id)
             return {}
+
+    def _qga_call_get_vcpus(self, vm):
+        guest_info = {'guestCPUCount': -1}
+        try:
+            self.log.debug('Requesting vCPU info for vm=%s', vm.id)
+            with vm.qga_context(_COMMAND_TIMEOUT):
+                vcpus = QemuGuestAgentDomain(vm).guestVcpus()
+        except (exception.NonResponsiveGuestAgent, libvirt.libvirtError) as e:
+            self.log.info('Failed to get guest info for vm=%s, error: %s',
+                          vm.id, e)
+            self.set_failure(vm.id)
+            return {}
+        except virdomain.NotConnectedError:
+            self.log.debug(
+                'Not querying QEMU-GA because domain is not running ' +
+                'for vm-id=%s', vm.id)
+            return {}
+        if 'online' in vcpus:
+            guest_info['guestCPUCount'] = len(taskset.cpulist_parse(
+                vcpus['online']))
```

Comment:
``` python
online_cpus = taskset.cpulist_parse(vcpus['online'])
guest_info['guestCPUCount'] = len(online_cpus)
```

would look better.

Or even better:

``` python
if 'online' in vcpus:
    count = len(taskset.cpulist_parse(vcpus['online']))
else:
    count = -1
return {'guestCPUCount': online}
```

(and dropping the line at the beginning of the method).